### ISSUE TITLE
remove unconditional gnureadline dependency

### ISSUE THREAD

**Issue by ivanov** [ipython/rlipython#16]
This is only really necessary on OS X, but the code to express that it a bit finicky, so I'll just include it unconditionally for now.

**Comment by ivanov** [comment]
eek... maybe this was a bad idea... 

```
(cl) 23:58@rlipython(master)$ ipython
Python 3.6.0 (default, Jan 18 2017, 19:21:48) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.0.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: 
python3.6(5000,0x104f57310) malloc: *** error for object 0x105f5bad0: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
```

**Comment by ajsteven130** [comment]
This is also a problem on Windows where GNUreadline just won't install. It suggests using pyreadline instead, but I already have that installed.

**Comment by braham-snyder** [comment]
workaround mentioned in that issue (working on El Capitan at least--master branch works, PyPI's version does not):
```
pip install https://github.com/ludwigschwardt/python-gnureadline/archive/master.zip --upgrade
```

**Comment by ivanov** [comment]
Ok, this is now fixed as of ba246dea1f1b249a3f981a8de24989db310ffa71

Thanks @braham-snyder and @ajsteven130  for pitching in. 
Happy hacking! :bowtie:
